float32 old_speed
float32 old_baseline
float32 current_speed
float32 current_baseline

float32 old_mass
float32 current_mass